### DISCUSSION TITLE
DataGrid scroll paging is too small

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#15968] [Q&A] [answered]
When I click the vertical scrollbar track of a DataGrid, I expect a larger scroll, but that is not happening.
I can't find anywhere to set the 'large change' property.
Here is my xaml:
```

			<DataGrid Name="grdTest"
						x:DataType="vm:FooViewModel"
						ItemsSource="{Binding _items}"
						SelectionMode="Single"
						HeadersVisibility="None"
						Height="173"
						VerticalAlignment="Stretch"
						HorizontalScrollBarVisibility="Hidden"
						BorderThickness="1" BorderBrush="Black" Margin="3">

				<DataGrid.Columns>
					<DataGridCheckBoxColumn Binding="{Binding _selected}" Width="0.05*"/>
					<DataGridTextColumn Binding="{Binding _name}" IsReadOnly="True" Width="0.95*"/>
				</DataGrid.Columns>
			</DataGrid>
```

**Comment by Pratham95VP** [comment]
Unlike WPF, Avalonia's DataGrid does not have a built-in ScrollViewer property to control the scrolling behavior. By wrapping the `DataGrid` inside a` ScrollViewer`, you should be able to achieve the desired scrolling behavior.

**Reply by timunie** [reply, marked as answer]
No you shouldn't do that. This breaks virtualization and Layout! 

You may be able to use a Style selector for this:
https://github.com/AvaloniaUI/Avalonia/blob/82e3613939af5451597d3c19f3a48e0a8ba888f3/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L551-L555

Something like that (untested):

```xml
<Style Selector="DataGrid /template/ ScrollBar#PART_VerticalScrollbar">
...
</Style>
```

**Reply by Pratham95VP** [reply]
Thank you for the correction. Could you please provide an explanation as to why wrapping the DataGrid inside a ScrollViewer might disrupt virtualization and layout functionalities?

**Reply by timunie** [reply]
It needs fixed bounds to caculate which items to materialize. If you have inifinite space, that doesn't work correctly.

**Reply by Pratham95VP** [reply]
got it. Thank you.
